tests/instrumentation/psycopg_tests.py

```diff
@@ -0,0 +1,279 @@
+        assert span["context"]["service"]["target"]["name"] == "elasticapm_test"
+
+
+@pytest.mark.integrationtest
+@pytest.mark.skipif(not has_postgres_configured, reason="PostgresSQL not configured")
+def test_psycopg_composable_query_works(instrument, postgres_connection, elasticapm_client):
+    """
+    Check that we parse queries that are psycopg.sql.Composable correctly
+    """
+    from psycopg import sql
+
+    cursor = postgres_connection.cursor()
+    query = sql.SQL("SELECT * FROM {table} WHERE {row} LIKE 't%' ORDER BY {row} DESC").format(
+        table=sql.Identifier("test"), row=sql.Identifier("name")
+    )
+    baked_query = query.as_string(cursor.__wrapped__)
+    result = None
+    try:
+        elasticapm_client.begin_transaction("web.django")
+        cursor.execute(query)
+        result = cursor.fetchall()
+        elasticapm_client.end_transaction(None, "test-transaction")
+    finally:
+        # make sure we've cleared out the spans for the other tests.
+        assert [(2, "two"), (3, "three")] == result
+        transactions = elasticapm_client.events[TRANSACTION]
+        spans = elasticapm_client.spans_for_transaction(transactions[0])
+        span = spans[0]
+        assert span["name"] == "SELECT FROM test"
+        assert "db" in span["context"]
+        assert span["context"]["db"]["instance"] == "elasticapm_test"
+        assert span["context"]["db"]["type"] == "sql"
+        assert span["context"]["db"]["statement"] == baked_query
+
+
+@pytest.mark.integrationtest
+@pytest.mark.skipif(not has_postgres_configured, reason="PostgresSQL not configured")
+def test_psycopg_binary_query_works(instrument, postgres_connection, elasticapm_client):
+    """
+    Check that we pass queries with %-notation but without parameters
+    properly to the dbapi backend
+    """
+    cursor = postgres_connection.cursor()
+    query = b"SELECT * FROM test WHERE name LIKE 't%'"
+
+    baked_query = query.decode()
+    try:
+        elasticapm_client.begin_transaction("web.django")
+        cursor.execute(query)
+        result = cursor.fetchall()
+        elasticapm_client.end_transaction(None, "test-transaction")
+    finally:
+        # make sure we've cleared out the spans for the other tests.
+        assert [(2, "two"), (3, "three")] == result
+        transactions = elasticapm_client.events[TRANSACTION]
+        spans = elasticapm_client.spans_for_transaction(transactions[0])
+        span = spans[0]
+        assert span["name"] == "SELECT FROM test"
+        assert "db" in span["context"]
+        assert span["context"]["db"]["instance"] == "elasticapm_test"
+        assert span["context"]["db"]["type"] == "sql"
+        assert span["context"]["db"]["statement"] == baked_query
+
+
+@pytest.mark.integrationtest
+@pytest.mark.skipif(not has_postgres_configured, reason="PostgresSQL not configured")
+def test_psycopg_call_stored_function(instrument, postgres_connection, elasticapm_client):
```

Comment:
`psycopg2_tests.py` has [a similar test](https://github.com/minhtule/apm-agent-python/blob/35acac150833e5b451a82a276d892dd6f6c75c28/tests/instrumentation/psycopg2_tests.py#L427-L447) to test [`cursor.callproc()`](https://www.psycopg.org/docs/cursor.html#cursor.callproc). However, Psycopg3 removes this method and recommends to use `cursor.execute()` ([doc](https://www.psycopg.org/psycopg3/docs/basic/from_pg2.html#callproc-is-gone)) so I'm not really sure if we should keep this test.

Reply:
I think that's reasonable, we can skip that test.

Reply:
Thank you for confirming! I've removed this test.